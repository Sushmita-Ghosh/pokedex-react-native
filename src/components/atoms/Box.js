import React, { useState } from "react";
import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
import { map, capitalize, truncate } from "lodash";
import getColorByPokemonType from "../../utils/getColorByPokemonType";

export default function Box(props) {
  const { data, isWeak, isModal, setTypesArray } = props;

  const [pressed, setPressed] = useState(false);

  const getPokemonTypes = (item) => {
    setPressed(true);
    console.log(item);
    if (item) {
      // (prevTypes) => [...prevTypes, item]
      setTypesArray(item);
    }
  };

  // const getPillColor = (item) =>
  //   pressed
  //     ? getColorByPokemonType("unknown")
  //     : getColorByPokemonType(!isWeak ? item.type.name : item.name);

  return (
    <View style={styles.content}>
      {data &&
        data.map((item, index) => (
          <View
            key={index}
            style={{
              ...styles.pill,
              backgroundColor: getColorByPokemonType(
                !isWeak ? item.type.name : item.name
              ),
            }}
          >
            {isModal ? (
              <TouchableOpacity
                // style={{
                //   width: "100%",
                //   backgroundColor: "red",
                // }}
                onPress={() => getPokemonTypes(item)}
              >
                <Text style={styles.type}>
                  {capitalize(!isWeak ? item.type.name : item.name)}
                </Text>
              </TouchableOpacity>
            ) : (
              <Text style={styles.type}>
                {capitalize(!isWeak ? item.type.name : item.name)}
              </Text>
            )}
          </View>
        ))}
    </View>
  );
}

const styles = StyleSheet.create({
  content: {
    flexDirection: "row",
    // alignItems: "center",
    // justifyContent: "center",
    flexWrap: "wrap",
    width: "100%",
  },
  pill: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: 10,
    margin: 2,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "solid",
    // height: 40,
  },
  type: {
    color: "black",
    fontSize: 15,
  },
});
