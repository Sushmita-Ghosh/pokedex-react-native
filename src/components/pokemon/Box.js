import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { map, capitalize } from "lodash";
import getColorByPokemonType from "../../utils/getColorByPokemonType";

export default function Box(props) {
  const { data, isWeak } = props;

  return (
    <View style={styles.content}>
      {data &&
        data.map((item, index) => (
          <View
            key={index}
            style={{
              ...styles.pill,
              backgroundColor: getColorByPokemonType(
                !isWeak ? item.type.name : item.name
              ),
            }}
          >
            <Text style={styles.type}>
              {capitalize(!isWeak ? item.type.name : item.name)}
            </Text>
          </View>
        ))}
    </View>
  );
}

const styles = StyleSheet.create({
  content: {
    flexDirection: "row",
    // alignItems: "center",
    // justifyContent: "center",
    flexWrap: "wrap",
    width: "100%",
  },
  pill: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    borderRadius: 10,
    margin: 2,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "solid",
  },
  type: {
    color: "black",
    fontSize: 15,
  },
});
