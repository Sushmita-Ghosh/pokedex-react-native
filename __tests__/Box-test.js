import "react-native";
import React from "react";
import Box from "../src/components/pokemon/Box.js";
import renderer from "react-test-renderer";

test("render app snapshot", () => {
  const snapshot = renderer.create(<Box />).toJSON();

  expect(snapshot).toMatchSnapshot();
});
