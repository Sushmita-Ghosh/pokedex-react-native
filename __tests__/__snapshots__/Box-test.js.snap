// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "width": "100%",
    }
  }
/>
`;
